crypto/src/main/kotlin/net/corda/v5/crypto/DigitalSignedDocument.kt

```diff
@@ -0,0 +1,16 @@
+package net.corda.v5.crypto
+
+import net.corda.v5.base.annotations.CordaSerializable
+
+
+@CordaSerializable
+/** A representation of a signed document,
+ *
+ * @param payload the payload that is to be signed, which must be Corda serializable
+ * @param metadata additional records. Needs to be Corda serializable..
+ */
+class DigitalSignedDocument<T, M>(
+    val payload: T,
```

Comment:
Based on this design proposal, `DigitalSignedDocument` is what `sign` produces and what `verify` consumes and that makes the two APIs symmetrical. However, my understanding is that our main use case for `sign` and `verify` APIs is:

A has a payload to be signed by B, then it sends it across the wire to B, and B signs it and sends back to A its signature to be verified by A who already possess the payload that was signed, so why B send back again the payload itself. This might also means if B starts acting maliciously, and A was lazy enough to not compare its payload with the payload that was sent back to be the same, B could have changed the payload over which they signed. 

Well, we could be recreating `DigitalSignedDocument` to be passed in on verify side with the payload A holds and then maybe B just sends over only the signature from this structure, but this structure pretty much implies to me at least that it gets returned from `sign` operation to be passed in to `verify` operation.

Less importantly, it also means overhead to the wire.

So maybe we should not be including the payload here? Or will there be a need for us to include the payload as well here? @mnesbit @lankydan what do you think?